pkg/dashboard/functiontemplates/githubfunctiontemplatefetcher.go

```diff
@@ -0,0 +1,239 @@
+/*
+Copyright 2017 The Nuclio Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package functiontemplates
+
+import (
+	"context"
+	"encoding/base64"
+	"strings"
+
+	"github.com/nuclio/nuclio/pkg/errors"
+
+	"github.com/google/go-github/github"
+	"github.com/icza/dyno"
+	"golang.org/x/oauth2"
+	"gopkg.in/yaml.v2"
+)
+
+type GithubFunctionTemplateFetcher struct {
+	branch          string
+	owner           string
+	repository      string
+	githubAPIClient *github.Client
+	FunctionTemplateFetcher
```

Comment:
Why are you embedding this? You comply to the interface, you don't need to declare that